client.lua

```diff
@@ -1,58 +1,67 @@
-local isSongPlaying = false
- 
-RegisterNetEvent('3dsound:playSound')
-AddEventHandler('3dsound:playSound',function(audioSource, audioCoord, audioRotation)
-	local ped = PlayerPedId()
-	
-	-- Send NUI message to start the audio
-	SendNUIMessage({transactionType = 'start',audioCoord = audioCoord,
-		audioRot=audioRotation, audioSource = audioSource})
-	
-	isSongPlaying = true  -- Set song playing to true 
-	
+local songs = {}
 
-	Citizen.CreateThread(function()
-		while isSongPlaying do
-			Wait(100)
-			pCoord = GetEntityCoords(ped)
-			-- prot = GetGameplayCamRot(0)  -- Uncomment this to use player camera coord
-			pRot = GetEntityRotation(ped)
-			if #(audioCoord-pCoord) < 30 then
-				SendNUIMessage({transactionType = 'update',
-					pedCoord = pCoord, pedRot = pRot, mute = false})
-			else
-				SendNUIMessage({transactionType = 'update',
-					pedCoord = pCoord, pedRot = pRot, mute = true})
+Citizen.CreateThread(function()
+	while true do
```

Comment:
I think this thread can be intiated inside the event itself. We really don't need to create it at the start and keep it running even when no sound is active. 
It should be done inside the event, as done in the original code with `while songs[uid].isSongPlaying do` as the condition

Reply:
I disagree, for example - if multiple sound events were triggered to the client, this would be sending the coordinates multiple times in separate threads for each event whereas it could all be done centrally through one thread.

Reply:
Well that can be solved easily. We can have a global variable `isUpdatingCoord` which can be set to `true` everytime we are updating player coord, and before before creating a thread we just need to do `if not isUpdatingCoord then` and create a thread else we just skip creating a thread and just send NUIMessage to play the sound